### PR TITLE
Torpedo Warhead IED rework

### PR DESCRIPTION
**Pull request by Andromeda-K22**
Urgh. Balancing pass, yay.

Still needs some work - minor stuff, plus whatever is found in reviews. I need to stop opening these WIP PRs.

Currently:

You now have to hack torpedo warheads, and find the correct wire to DEACTIVATE the safeties.

To reactivate them, you must find ANOTHER wire, cut it, then pulse the one that deactivates the safeties.

Pulse the wrong with with safeties off? BOOM.

Cut the wrong wire with safeties on? BOOM.

### CODE REVIEW

File: code/modules/urist/modules/shipbattles/torpedo_warhead.dm [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+	..(user)
+	switch(riggedstate)
+		if(CIRCUITRY_EXPOSED) to_chat(user, "<span class='notice'>It's control circuitry is exposed.</span>")
+		if(CIRCUITRY_MODIFIED) to_chat(user, "<span class='notice'>It's control circuitry is exposed, and the internal wiring appears to have been modified.</span>")
+		if(CIRCUITRY_RIGGED) to_chat(user, "<span class='notice'>It's control circuitry is exposed, the internal wiring appears to have been modified, and a wire has been rigged to \The [attached_device].</span>")
+	switch(safety)
+		if(0) to_chat(user, "<span class='warning'>The safeties have been disabled.</span>")
+		if(1) to_chat(user, "<span class='notice'>The safeties are enabled.</span>")
+	if(is_rigged)
+		to_chat(user, "<span class='warning'>There is a [attached_device] attached to the warhaed.</span>")
```
Comment (Naxodile) [inline code/modules/urist/modules/shipbattles/torpedo_warhead.dm]:
```suggestion
		to_chat(user, "<span class='warning'>There is \a [attached_device] attached to the warhead.</span>")
```

File: code/modules/urist/modules/shipbattles/torpedo_warhead.dm [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+/obj/item/shipweapons/torpedo_warhead/proc/detonate(var/forced = 0)
+	playsound(src.loc, 'sound/machines/buttonbeep.ogg', 25, 0, 10)
+	if(safety && !forced)
+		visible_message("<span class='danger'>[src] beeps stubbornly, refusing to detonate!</span>")
+		playsound(src.loc, 'sound/machines/buzz-sigh.ogg', 25, 0, 10)
+	if(forced)
+		visible_message("<span class='danger'>[src] pings, begining a short countdown!</span>")
+		playsound(src.loc, 'sound/machines/ping.ogg', 25, 0, 10)
+		playsound(src.loc, 'sound/items/countdown.ogg', 25, 0, 10)
+		spawn(4 SECONDS)
```
Comment (Naxodile) [inline code/modules/urist/modules/shipbattles/torpedo_warhead.dm]: There's now the rare possibility that one could re-engage the safeties _after_ its set to explode. If you're feeling kind you could give the madman not running away a chance at survival.

File: code/modules/urist/modules/shipbattles/torpedo_warhead.dm [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+	var/obj/item/device/attached_device //Whatever is attached that'll detonate us.
+	var/datum/wires/torpedowarhead/wires = null
+
+/obj/item/shipweapons/torpedo_warhead/New()
+	..()
+	wires = new/datum/wires/torpedowarhead(src)
+
+/obj/item/shipweapons/torpedo_warhead/Destroy()
+	qdel(wires)
+	wires = null
```
[the remark this answers is not in the document]
Comment (Naxodile) [reply code/modules/urist/modules/shipbattles/torpedo_warhead.dm]: Same thing for attached_device.

File: code/modules/urist/modules/shipbattles/torpedo_warhead.dm [hunk 4]
```diff
@@ -0,0 +1,194 @@
+//TORPEDO WARHEAD BEGIN//
+//Torpedo IEDs begin
+#define PANEL_CLOSED 1
```
Comment (Naxodile) [inline code/modules/urist/modules/shipbattles/torpedo_warhead.dm]: Unused define.

File: code/modules/urist/modules/shipbattles/torpedo_warhead.dm [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+			playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+			icon_state = "torpedowarhead-open-mod-armed"
+	return
+
+/obj/item/shipweapons/torpedo_warhead/attack_self(mob/user as mob)
+	..()
+	if(riggedstate < CIRCUITRY_EXPOSED)
+		wires.Interact(user)
+
+/obj/item/shipweapons/torpedo_warhead/proc/process_activation() // uh oh, time to boom
```
Comment (Naxodile) [inline code/modules/urist/modules/shipbattles/torpedo_warhead.dm]: I spy an unused function.

Comment (Andromeda-K22) [reply code/modules/urist/modules/shipbattles/torpedo_warhead.dm]: Process activation is used for signallers.

**Comment by Andromeda-K22** [comment]
@Glloyd Remove DNM please.

File: code/modules/urist/modules/shipbattles/torpedo_warhead.dm [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+
+/obj/item/shipweapons/torpedo_warhead/New()
+	..()
+	wires = new/datum/wires/torpedowarhead(src)
+
+/obj/item/shipweapons/torpedo_warhead/Destroy()
+	QDEL_NULL(wires)
+	wires = null
+	QDEL_NULL(attached_device)
+	attached_device = null
```
Comment (Naxodile) [inline code/modules/urist/modules/shipbattles/torpedo_warhead.dm]: Nulling variables is already handled by the QDEL_NULL.

**Comment by Glloyd** [comment]
@Naxodile are you happy with this? If so, I'll merge

**Pull request merged**